### PR TITLE
AUTH-2655 -initial cloud front resource creation

### PR DESCRIPTION
**Pull request by pskushwaha1**
## What

AUTH-2655 Initial cloud-front resource creation

Initial cloud-front resource creation for AUTH
This PR is to create a Cloud front resource for each Env with `cloudfront_auth_frontend_enabled = true flag`
Most of the resource for cloud front get created in region` us-east-1`
TLS certificate , DNS records for CF origin, SNS for Notification , WAF for Cloud front & Cloud-front distribution & monitoring

##This PR Will not do the Traffic Switch will only create parallel resource for Cloud Front 

## How to Review
Code Review
Deploy to sandpit with ./deploy-sandpit.sh -t -p 
See if Cloud-front is deployed by checking on AWS console

### CODE REVIEW

File: ci/terraform/cloudfront.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  count        = var.cloudfront_auth_frontend_enabled ? 1 : 0
+  provider     = aws.cloudfront
+  name         = "${var.environment}-auth-fe-cloudfront-monitoring"
+  template_url = "https://template-storage-templatebucket-1upzyw6v9cs42.s3.amazonaws.com/cloudfront-monitoring-alarm/template.yaml?z=${timestamp()}"
+
+  capabilities = ["CAPABILITY_NAMED_IAM"]
+
+  parameters = {
+    CacheHitAlarmSNSTopicARN            = aws_sns_topic.slack_events[0].arn
+    CloudFrontAdditionaldMetricsEnabled = false
```
Comment (ethanmills) [inline ci/terraform/cloudfront.tf:37]: I think this should be true? Otherwise it won't monitor cache hit rate

Comment (ethanmills) [inline ci/terraform/cloudfront.tf:37]:
```suggestion
    CloudFrontAdditionalMetricsEnabled = false
```

File: ci/terraform/sandpit.tfvars [hunk 2]
```diff
@@ -15,10 +16,16 @@ support_authorize_controller  = "1"
...
 
+
 frontend_task_definition_cpu     = 256
 frontend_task_definition_memory  = 512
 frontend_auto_scaling_v2_enabled = true
 
+#cloudfront varaibles
+cloudfront_auth_frontend_enabled    = true
+auth_origin_cloakingheader          = "dsxUBpHUdcOYUBICSm3LuxA4V6yKGkz0"
+previous_auth_origin_cloakingheader = "dsxUBpHUdcOYUBICSm3LuxA4V6yKGkz0"
```
Comment (ethanmills) [inline ci/terraform/sandpit.tfvars:27]: These should be considered secret

File: ci/terraform/sandpit.tfvars [hunk 3]
```diff
@@ -15,10 +16,16 @@ support_authorize_controller  = "1"
 support_account_interventions = "1"
 support_2fa_b4_password_reset = "1"
 
+
 frontend_task_definition_cpu     = 256
 frontend_task_definition_memory  = 512
 frontend_auto_scaling_v2_enabled = true
 
+#cloudfront varaibles
```
Comment (ethanmills) [inline ci/terraform/sandpit.tfvars:24]:
```suggestion
#cloudfront variables 
```

File: ci/terraform/sns.tf [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      "SNS:GetTopicAttributes",
+      "SNS:DeleteTopic",
+      "SNS:AddPermission",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
```
Comment (ethanmills) [inline ci/terraform/sns.tf:34]:
```suggestion
      type        = "Service"
      identifiers = ["cloudwatch.amazonaws.com"]
```
Does this work instead? I think this topic policy gives anyone in the world permission to delete the topic?

File: ci/terraform/sns.tf [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    }
```
Comment (ethanmills) [inline ci/terraform/sns.tf:35]:
```suggestion
 principals {
      type        = "Service"
      identifiers = ["cloudwatch.amazonaws.com"]
    }

    condition {
      test     = "StringEquals"
      values   = [data.aws_caller_identity.current.account_id]
      variable = "aws:SourceAccount"
    }
```
Does this work instead? I think this topic policy gives anyone in the world permissions?

File: ci/terraform/cloudfront.tf (hunk 1 above)
Comment (pskushwaha1) [reply ci/terraform/cloudfront.tf:37]: done

File: ci/terraform/sandpit.tfvars (hunk 2 above)
Comment (pskushwaha1) [reply ci/terraform/sandpit.tfvars:27]: how do we mange secret for env which are deployed locally ? I can implemented that

File: ci/terraform/cloudfront.tf [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    CloudfrontWafAcl               = aws_wafv2_web_acl.frontend_cloudfront_waf_web_acl[0].arn
+    DistributionAlias              = local.frontend_fqdn
+    FraudHeaderEnabled             = var.Fraud_Header_Enabled
+    OriginCloakingHeader           = var.auth_origin_cloakingheader
+    OriginResourceArn              = aws_lb.frontend_alb.id
+    OriginWafAcl                   = "none"
+    PreviousOriginCloakingHeader   = var.previous_auth_origin_cloakingheader
+    StandardLoggingEnabled         = true
+  }
+  depends_on = [aws_acm_certificate.cloudfront_frontend_certificate, aws_wafv2_web_acl.frontend_cloudfront_waf_web_acl, aws_lb.frontend_alb]
```
[the remark this answers is not in the document]
Comment (pskushwaha1) [reply ci/terraform/cloudfront.tf:22]: removed  depends_on

File: ci/terraform/sandpit.tfvars (hunk 2 above)
Comment (ethanmills) [reply ci/terraform/sandpit.tfvars:27]: Ah this repo doesn't have a read secrets script in the deployment like api 😞

**Review by dbes-gds — APPROVED** [review]

**Pull request merged**